sessiontxn/isolation/readcommitted.go

```diff
@@ -230,14 +282,54 @@ func (p *PessimisticRCTxnContextProvider) AdviseWarmup() error {
 	return nil
 }
 
+func planSkipGetTsoFromPD(sctx sessionctx.Context, plan plannercore.Plan, inLockOrWriteStmt bool) bool {
```

Comment:
A bit confused about why we need two different functions `NeedDisableWarmupInOptimizer` and `planSkipGetTsoFromPD` which have similar plan check logic. Could `use_last_check` be checked and set once and then the constant ts future is always used unless the last ts is updated?

Reply:
The function `NeedDisableWarmupInOptimizer` is called in `OnStmtStart`, The execution plan hasn't been made at that time. we can't decide whether to get ts from PD at this time.
`planSkipGetTsoFromPD` is called in `AdviseOptimizeWithPlan`, the execution plan has made at this time.

Reply:
The best way is as bellow, I do it as bellow at begin. But this proposal can't decrease tso request count.
Another optimize way is that we don't disable AdviseWarmup , which means optimizer sends tso request as before, but we don't wait this tso and use latest tso in function AdviseOptimizeWithPlan just like what the insert statements do as before. This new proposal doesn't decrease tso requests but decrease tso wait time.

Reply:
Could the current implementation decrease the effect of warmup if the statement can't skip getting tso finally?

Reply:
Yes, I already mentioned it in issue talk.